gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/TXState.java

```diff
@@ -1159,6 +1160,49 @@ public boolean execute(Object key) {
     return eventsToFree;
   }
 
+  private void publishRecordedVersions() {
+    final TXRegionState[] finalRegions = this.finalizeRegions;
+    GemFireCacheImpl cache = GemFireCacheImpl.getExisting();
+    final LogWriterI18n logger = getTxMgr().getLogger();
+    // No need to check for snapshot if we want to enable it for RC.
+    if (cache.snapshotEnabled()) {
```

Comment:
This check can be called before calling publishRecordedVersions. If snapshot is not enabled then this can be avoided. There are a lot of method calls like isSnapshot, snapShotEnabled then there is a flag called DEFAULT_SNAPSHOT_ENABLED_TX which I feel are doing duplicate kind of work of checking snapshot enabled etc with little differences. At some point we should do this cleanup especially for row tables it will become important for low latency.